lib/spree/retailops/rop_order_helper.rb

```diff
@@ -141,7 +141,7 @@ def calculate_ship_price(removed = false)
         # This resolves https://github.com/dotandbo/dotandbo-spree/issues/3857
         # Where entire shipping fee is removed from order if 
         # last unshipped line_item is canceled.
-        if removed && !method && @order.shipments.present? && !@order.shipments.unshipped.present?
+        if removed && !method && @order.shipments.present? && @order.shipments.where('state != ?','shipped').empty?
```

Comment:
I noticed that we aren't really using `unshipped` scope anywhere else, so I thought adding a new scope just for this rare case was kind of redundant. I agree that adding the scope may be the cleaner fix. Let me know what you think and I will adjust accordingly. @sankalpk